junit-platform-reporting/src/main/java/org/junit/platform/reporting/legacy/xml/XmlReportWriter.java

```diff
@@ -51,8 +52,7 @@
  */
 class XmlReportWriter {
 
-	private static final String CDATA_START = "<![CDATA[";
-	private static final String CDATA_END = "]]>";
+	private static final Pattern CDATA_SPLIT_PATTERN = Pattern.compile("(?<=]])(?=>)");
```

Comment:
❓ Why is this pattern so complicated? Is the positive lookbehind and the positive lookahead necessary?

Reply:
And if the the positive lookbehind and  positive lookahead are necessary, please add comments in the code to explain exactly what the pattern matches and why.

Reply:
@chschu Could you please take a look? We'd like to include this in 5.6 M1.

Reply:
Sorry for the delay.

The zero-width assertions are used to simplify the loop in `writeCDataSafely(...)`. Any characters captured by the pattern itself would have to be added to the results of `String.split(...)`.

I added a comment that should clarify this.

Reply:
Thanks! 👍